### ISSUE TITLE
rust-gmp-0.2.10 beta regression

### ISSUE THREAD

**Issue by est31** [rust-lang/rust#43679]
[rust-gmp](https://crates.io/crates/rust-gmp) version 0.2.10 regressed from stable to beta ([full log](https://cargobomb-reports.s3.amazonaws.com/stable-1.19-beta-1.20.0-beta.1/beta-2017-07-18/reg/rust-gmp-0.2.10/log.txt)):

```
Aug 03 19:34:33.891 INFO kablam! error: type `rand::gmp_randstate_struct` is private
Aug 03 19:34:33.891 INFO kablam!   --> src/mpz.rs:91:5
Aug 03 19:34:33.891 INFO kablam!    |
Aug 03 19:34:33.891 INFO kablam! 91 |     fn __gmpz_urandomm(rop: mpz_ptr, state: gmp_randstate_t, n: mpz_srcptr);
Aug 03 19:34:33.891 INFO kablam!    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Aug 03 19:34:33.891 INFO kablam! 
```
cc @fizyk20

**Comment by arielb1** [comment]
- #42125 - more strict privacy rules.

Fixed in 0.5.0

**Comment by alexcrichton** [comment]
I believe the compiler/lang teams won't be reverting https://github.com/rust-lang/rust/pull/42125, so I'm going to close this regression tracker.
